draft-ietf-httpbis-priority.md

```diff
@@ -108,101 +114,100 @@ The term control stream is used to describe the HTTP/2 stream with identifier
+instead favor custom server-driven schemes based on heuristics or other hints,
+such as resource content type or request generation order. For example, a
+server, with knowledge of the document structure, might want to prioritize the
+delivery of images that are critical to user experience above other images, but
+below the CSS files. Since client trees vary, it is impossible for the server to
+determine how such images should be prioritized against other responses.
+
+RFC 7540 allows intermediaries to coalesce multiple client trees into a
 single tree that is used for a single upstream HTTP/2 connection. However, most
-intermediaries do not support this. The scheme does not define a method that can
+intermediaries do not support this. RFC 7540 does not define a method that can
 be used by a server to express the priority of a response. Without such a
 method, intermediaries cannot coordinate client-driven and server-driven
 priorities.
 
-HTTP/2 describes denial-of-service considerations for implementations. On
+RFC 7540 describes denial-of-service considerations for implementations. On
 2019-08-13 Netflix issued an advisory notice about the discovery of several
-resource exhaustion vectors affecting multiple HTTP/2 implementations. One
-attack, [CVE-2019-9513] aka "Resource Loop", is based on manipulation of the
-priority tree.
-
-The HTTP/2 scheme depends on in-order delivery of signals, leading to challenges
-in porting the scheme to protocols that do not provide global ordering. For
-example, the scheme cannot be used in HTTP/3 {{HTTP3}} without
-changing the signal and its processing.
-
-Considering the problems with deployment and adaptability to HTTP/3, retaining
-the HTTP/2 priority scheme increases the complexity of the entire system without
-any evidence that the value it provides offsets that complexity. In fact,
-multiple experiments from independent research have shown that simpler schemes
-can reach at least equivalent performance characteristics compared to the more
-complex HTTP/2 setups seen in practice, at least for the web use case.
-
-## Disabling HTTP/2 Priorities {#disabling}
-
-The problems and insights set out above are motivation for allowing endpoints to
-opt out of using the HTTP/2 priority scheme, in favor of using an alternative
-such as the scheme defined in this specification. The
-SETTINGS_DEPRECATE_HTTP2_PRIORITIES setting described below enables endpoints to
-understand their peer's intention. The value of the parameter MUST
-be 0 or 1. Any value other than 0 or 1 MUST be treated as a connection error
-(see {{Section 5.4.1 of HTTP2}}) of type PROTOCOL_ERROR.
+resource exhaustion vectors affecting multiple RFC 7540 implementations. One
+attack, [CVE-2019-9513] aka "Resource Loop", is based on using priority signals
+to manipulate the server's stored prioritization state.
+
+HTTP/2 priority signals depend on in-order delivery, leading to challenges in
```

Comment:
imo, this is vague... in-order delivery of what? the signals? requests? responses? 

```suggestion
HTTP/2 priority signals depend on their in-order delivery to deterministically construct the intended tree at the receiver, leading to challenges in
```